### PR TITLE
ci: improve developer-local docs and make fastbuild envoy binary acce…

### PR DESCRIPTION
**Pull request by htuch**
…ssible.

### CODE REVIEW

**Comment by htuch** [comment]
@moderation

**Comment by htuch** [comment]
Can you hold off on merge, I have a small change suggested by @PiotrSikora

**Comment by htuch** [comment]
@PiotrSikora @moderation

File: ci/README.md [hunk 1]
```diff
@@ -20,21 +20,36 @@ master commit at which the binary was compiled, and `latest` corresponds to a bi
...
+The Envoy binary can be found in `/tmp/envoy-docker-build/envoy/source/exe/envoy-fastbuild` on the Docker host. You
+can control this by setting `ENVOY_DOCKER_BUILD_DIR` in the environment, e.g. to
+generate the binary in `~/build/envoy/source/exe/envoy-fastbuild` you can run:
+
+
+```bash
+ENVOY_DOCKER_BUILD_DIR=~/build ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev.server_only'
+```
+
+For a release Envoy binary you can run:
```
Comment (PiotrSikora) [inline ci/README.md]: `s/release Envoy binary/release version of Envoy binary/`?

File: ci/README.md [hunk 2]
```diff
@@ -20,21 +20,36 @@ master commit at which the binary was compiled, and `latest` corresponds to a bi
...
 
-The `./ci/run_envoy_docker.sh './ci/do_ci.sh <TARGET>'` targets are:
 
 * `bazel.asan` &mdash; build and run tests under `-c dbg --config=asan`.
-* `bazel.fastbuild` &mdash; build Envoy static binary under `-c fastbuild`.
-* `bazel.fastbuild.test` &mdash; build and run tests under `-c fastbuild`.
+* `bazel.dev.server_only` &mdash; build Envoy static binary under `-c fastbuild`.
+* `bazel.dev.test` &mdash; build and run tests under `-c fastbuild`.
 * `bazel.release` &mdash; build Envoy static binary and run tests under `-c opt`.
+* `bazel.release.server_only` &mdash; build Envoy static binary under `-c opt`.
```
Comment (PiotrSikora) [inline ci/README.md]: Why do we have `bazel.dev.test`, but `bazel.release` (without `.test`)?

**Comment by mattklein123** [comment]
LGTM will wait for @PiotrSikora +1

File: ci/README.md (hunk 2 above)
Comment (htuch) [inline ci/README.md]: Because the `bazel.release` target is used for the CI and combines the static binary build and tests (for performance reasons). For developers, you want a target that just builds/runs the tests (not as useful in CI). I can add a `bazel.release.tests` that only does the tests if you think it would be useful to developers.

I feel that so far, I'm hearing from folks like @moderation that they'd like the ability to build using the Docker image, but not for development work per se.

Comment (PiotrSikora) [inline ci/README.md]: Let me rephrase, we have:
1. `bazel.release.server_only` and `bazel.dev.server_only`, both building only static binary with  different `-c` options,
2. `baze.release` and `bazel.dev.test`, both building and running test with different `-c` options.

My question is why one has `.test` suffix, but not ther other.

I'm just nitpicking, it's not that important.

Comment (PiotrSikora) [inline ci/README.md]: (discussed this offline, @htuch is going to add binary build to `bazel.dev.test` and rename it to  `bazel.dev`)

File: ci/README.md [hunk 3]
```diff
@@ -20,21 +20,35 @@ master commit at which the binary was compiled, and `latest` corresponds to a bi
...
+For a release version of the Envoy binary you can run:
+
+```bash
+./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.release.server_only'
+```
+
+The artifacts can be found in `/tmp/envoy-docker-build/envoy/source/exe/envoy` (or wherever
+`$ENVOY_DOCKER_BUILD_DIR` points).
 
-The `./ci/run_envoy_docker.sh './ci/do_ci.sh <TARGET>'` targets are:
```
Comment (PiotrSikora) [inline ci/README.md]: Was this accidentally removed or on purpose?
